client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -200,8 +208,17 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        synchronizer.updateState((state, updates) -> {
+            updates.add(new UpdatingConfig(true));
+        });
+        getThrowingException(controller.updateReaderGroup(scope, groupName, config));
+        ReaderGroupConfig controllerConfig = getThrowingException(controller.getReaderGroupConfig(scope, groupName));
```

Comment:
Why is this call needed? Can it be combined with the previous one? (IE have it return this value)

Reply:
yes, i think there is a github issue to address this. 
ideally updateRG should return the generation number itself but it presently returns a boolean.

Reply:
The updateRG now returns the generation number which is used to update the state too.